wd_core.au3

```diff
@@ -1588,15 +1608,30 @@ Func __WD_Error($sWhere, $i_WD_ERROR, $sMessage = Default)
 				$sMsg = $sMsg & ": " & $sMessage
 			EndIf
 
+			$sMsg &= (($iExt) ? (' : Extended = ' & $iExt) : (''))

```

Comment:
I'm thinking this would be better if we showed the numeric value for both @error and @extended. Something like this (untested) --

`$sMsg &= '(' & $i_WD_ERROR & (($iExt) ? ' / ' &  $iExt) : '') & ')`

Reply:
Take a look: [EXT moved, added ERR numeric value](https://github.com/Danp2/au3WebDriver/pull/244/commits/def7c67a88f63ab16105ebe9638bd5ebaa3cd855)
What you think about such concept ?

Reply:
I would prefer something less wordy / intrusive. That's why I made the prior suggestion where the error codes were displayed in parentheses without further labeling. I even think we should take out the conditional display of @extended and always show the info like this --

`_WD_GetBrowserVersion ==> General Error (1 / 0)`

This improves the output of __WD_Error by always showing the error code, but it doesn't become too obtrusive.

Reply:
Take a look: [Func __WD_Error() - Refactoring](https://github.com/Danp2/au3WebDriver/pull/244/commits/4be951248cdadb096191db4f09d66427af7d0669)

Reply:
- What's point of adding $iErr and changing $i_WD_ERROR > $iErr? This doesn't appear to serve any real purpose.

- IIRC, the @compiled logic came over from FF.au3. I'm ok with removing this, but I wonder if it will cause a behavior change for existing scripts. Maybe `$_WD_ERROR_MSGBOX` should default to False?

- I would suggest further changing `WD_Core UDF Error` to `Webdriver UDF Error`

Reply:
> * What's point of adding $iErr and changing $i_WD_ERROR > $iErr? This doesn't appear to serve any real purpose.

I can revert it, if you wanted?

> * IIRC, the @compiled logic came over from FF.au3. I'm ok with removing this, but I wonder if it will cause a behavior change for existing scripts. Maybe `$_WD_ERROR_MSGBOX` should default to False?

Good idea about:
```AutoIt
Global $_WD_ERROR_MSGBOX = False ; Shows in compiled scripts error messages in msgboxes
```

Yes it change behavior. On compiled scirpt there was user interaction required when `MsgBox()` popups.
Now it will not be required by default. Unless sombody enable them.

I think that after all we need to suplement `wd_demo.au3` by adding this:
```autoit
_WD_Option("errormsgbox", (https://github.com/compiled =1))
_WD_Option("OutputDebug", (https://github.com/compiled =1))
```

> * I would suggest further changing `WD_Core UDF Error` to `Webdriver UDF Error`

Good point.

Reply:
as to `$iErr` and `$i_WD_ERROR`

```autoit
Func __WD_Error($sWhere, $i_WD_ERROR, $sMessage = Default, $iExt = 0)
	.......
	Local $sMsg, $iErr = $i_WD_ERROR ; as we set here $iErr = $i_WD_ERROR
	.......
	Switch $_WD_DEBUG
		.......
		Case $_WD_DEBUG_Error
			.......
		Case $_WD_DEBUG_Info
			.......
				If $_WD_ERROR_MSGBOX And $iErr < 6 Then
					.......
					If $iAnswer = $IDCANCEL Then
						$iErr = $_WD_ERROR_UserAbort ; evenutally change here
						.......
					EndIf
				EndIf
			.......
	EndSwitch

	Return $iErr ; and return here
EndFunc   ;==>__WD_Error
```

I was thinking that this is no more needed to use `$i_WD_ERROR`

After rethinking we can also use this function to not use `$i_WD_ERROR` variable at all and stick with commonly used name `$iErr`:

```autoit
Func __WD_Error($sWhere, $iErr, $sMessage = Default, $iExt = 0)
	.......
	Local $sMsg
	.......
	Switch $_WD_DEBUG
		.......
		Case $_WD_DEBUG_Error
			.......
		Case $_WD_DEBUG_Info
			.......
				If $_WD_ERROR_MSGBOX And $iErr < 6 Then
					.......
					If $iAnswer = $IDCANCEL Then
						$iErr = $_WD_ERROR_UserAbort ; evenutally change here
						.......
					EndIf
				EndIf
			.......
	EndSwitch

	Return $iErr ; and return here
EndFunc   ;==>__WD_Error
```

Reply:
> After rethinking we can also use this function to not use $i_WD_ERROR variable at all and stick with commonly used name $iErr

Yes, that's something that I had considered as well. IMO, this change isn't needed but it shouldn't hurt to change this.

Reply:
Take a look: [further refactoring](https://github.com/Danp2/au3WebDriver/pull/244/commits/046de6edf6a3404fdfb67e077d64aca85148d273)

Reply:
> `$iExt = 0`

I suggest that this be changed to `$iExt = Default` in the function declaration. Then you can distinguish between @extended truly equal to 0 versus not being passed from the calling routine.

Reply:
Take a look again on: [again refactoring](https://github.com/Danp2/au3WebDriver/pull/244/commits/ab2a021c3cdc14236114679ba73ea69408134b54)